pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -477,6 +479,62 @@ func (c *Controller) setupIndex() *AmbientIndexImpl {
 		}
 	})
 
+	c.configController.RegisterEventHandler(gvk.KubernetesGateway, func(oldCfg config.Config, newCfg config.Config, ev model.Event) {
```

Comment:
we will probably want to use `gatewayResourceClient` as done in https://github.com/istio/istio/pull/46501

Reply:
Would you consider this blocking or is this an improvement for a follow up PR?

Reply:
it can be a followup. The title is WIP, did you want to merge now?

Reply:
I was trying to move it in that direction however I recently noticed that it's unstable at startup and likely needs work. If an older istiod put a hostname into the address field it will panic (makes sense re: MustParse) but more worryingly in some instances the ztunnel will not be configured correctly and traffic will not be sent to the waypoint.

Reply:
That seems unrelated to using configController vs gatewayResourceClient?

the MustParse should be removed

Reply:
Swapped over to the gatewayResourceClient this morning and it looks to have resolved to startup bug. Running more tests but so far it is significantly more reliable on startup.

Reply:
Will push soon. I'm a bit uncertain about the initialization of that client which can now happen in 2 places depending on features enabled but that's hopefully something we can solve pretty quick or plan to iterate on.